lib/vyasa_web/components/contexts/discuss.ex

```diff
@@ -122,10 +176,202 @@ defmodule VyasaWeb.Context.Discuss do
     |> assign(sheaf_ui_lattice: nil)
   end
 
+  @doc """
+  Defines the drafting context by setting the draft_reflector_path and updating
+  the respective state and ui_state lattices.
+
+  Additionally, ensures that there's at least one draft mark in the marks that draft reflector
+  keeps in state.
+
+  There are no db writes that will be done in this function and anything downstream of it.
+  """
+  def init_drafting_context(
+        %Socket{
+          assigns: %{
+            session: %{sangh: %{id: _sangh_id}}
+          }
+        } = socket
+      ) do
+    socket
+    |> init_draft_reflector()
+    |> maybe_prepend_draft_mark_in_reflector()
+  end
+
+  # fallthrough
+  def init_drafting_context(%Socket{} = socket) do
+    socket
+    |> assign(draft_reflector_path: nil)
+  end
+
+  @doc """
+  Similar to the read mode, this function defines what the currently in-draft sheaf is for:
+  case 1: when there's a parent sheaf to it, it means that this draft sheaf, if published, will respond to that parent.
+  case 2: when there's no parent sheaf to it, then it means that the draft sheaf, if published, will be a root sheaf.
+
+  TODO: similar to the read::init_reply_to_context, url overrides may happen.
+  """
+  def init_reply_to_context(
```

Comment:
like in the read mode, we determine the reply to context by referring to the draft sheaf currently in the db and checking if there's an assoc-ed parent sheaf that might be a candidate for the reply_to context